File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/sample/AbstractOrder.java
package com.coupang.springframework.data.requery.domain.sample;

import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;

/**
 * AbstractOrder
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
@Table(name = "ORDERS")
public abstract class AbstractOrder extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    @ForeignKey
    @ManyToOne
    protected Customer customer;

    protected String orderNo;


    private static final long serialVersionUID = 3606365166891885684L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractOrder.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.*

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractOrder
 * @author debop
 * @since 18. 5. 30
 */
abstract class AbstractOrder: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract override val id: Long?

    @get:ManyToOne
    @get:Column(nullable = false)
    @get:ForeignKey(referencedColumn = "customer_id")
    abstract val customer: AbstractCustomer
}